tradingstrategy/candle.py

```diff
@@ -329,7 +329,13 @@ def get_candles_by_pair(self, pair_id: PrimaryKey) -> Optional[pd.DataFrame]:
 
             - close
         """
-        return self.get_samples_by_pair(pair_id)
+        if not hasattr(self, "candles_cache"):
+            self.candles_cache = {}
```

Comment:
Can you move this constructor?

It's cleaner to have it here and not to introduce new class variables in getter functions.